### ISSUE TITLE
Problem generating interpolation from Yale Faces B

### ISSUE THREAD

**Issue by kaihuchen** [zo7/deconvfaces#8]
I managed to find a copy of the _Extended Yale Faces Database B_ and trained a model using one single identity (just for testing) without problem. However, when I tried to generate some interpolated images I got the following error:

$ python faces.py generate -m output/FaceGen.YaleFaces.model.d5.adam.h5 -f params/interpolate.yaml -o generated
Using Theano backend.
Loading model...
Generating images...
----->parser.use_yale=True
----->num_images=121
----->inputs["identity"].shape=(121, 28)
----->inputs["identity"]=[[ 0.          1.          0.         ...,  0.          0.          0.        ]
 [ 0.          1.          0.         ...,  0.          0.          0.        ]
 [ 0.          1.          0.         ...,  0.          0.          0.        ]
 ...,
 [ 0.          0.59143859  0.         ...,  0.          0.          0.        ]
 [ 0.          0.58439443  0.         ...,  0.          0.          0.        ]
 [ 0.          0.57735027  0.         ...,  0.          0.          0.        ]]
  0%|                                                                                      | 0/2 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "faces.py", line 129, in <module>
    cmd()
  File "faces.py", line 96, in generate
    batch_size=args.batch_size, extension=args.extension)
  File "/mnt/mybk/tests/deconvfaces/faces/generate.py", line 680, in generate_from_yaml
    gen = model.predict_on_batch(batch)
  File "/home/kaihuchen01/anaconda2/envs/ph3/lib/python3.5/site-packages/keras/engine/training.py", line 1268, in predict_on_batch
    self.internal_input_shapes)
  File "/home/kaihuchen01/anaconda2/envs/ph3/lib/python3.5/site-packages/keras/engine/training.py", line 108, in standardize_input_data
    str(array.shape))
Exception: Error when checking : expected identity to have shape (None, 1) but got array with shape (64, 28)
$

Those prefixed with '----->' are what I printed out for debugging purposes. Any suggestion how I can get pass this? Thanks in advance!

**Comment by flynnmd** [comment]
The generate function is a really hacky thing I wrote quickly to be able to create animations, and the Yale database is really a tacked-on afterthought that I did to try to find an open dataset, so I'm really not sure what happened from this stack trace alone.

The yaml files have different parameters for Yale though -- instead of "or" (orientation) and "em" (emotion), it's "lt" (lighting) and "ps" (pose). I think I also hard-coded the number of identities in there, so if your dataset is missing people or has extra people it might crash.

**Comment by kaihuchen** [comment]
Finally got it to work. 

The following is a summary of what I had to do to make it work with the _Extended Yale Face Database B_:
1. As mentioned above the parameters file needs to be changed to reference only valid ids
2. The Yale DB uses different parameters (as mentioned above) so the parameters file needs to be updated accordingly.
3. Earlier I trained only one identity _yaleB30_ which then produces all kinds of error during the generation phase. I decided to switch to consecutive identities from the top, i.e., _yaleB01_ , _yaleB02_ , and _yaleB03_, the errors then magically went away.
4. It seems that the variable NUM_YALE_ID  in generate.py is not getting automatically updated to the total number of identities as expected. If I manually hard-code it (which should be 3 in my test case) then all's well.

Hope this helps somebody.

@zo7  Many thanks for the help.
